### PR TITLE
Configure ASB Prometheus

### PR DESCRIPTION
**Pull request by joheec**
# Type of PR

- [x] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

- template for Prometheus yaml
- Prometheus yaml for ngsadev
- Updated monitoring README with prometheus information

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes retaildevcrews/wcnp#320

### CODE REVIEW

File: deploy/ngsadev/monitoring/prometheus.yaml [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+kind: ConfigMap
+metadata:
+  name: prometheus-server-conf
+  labels:
+    name: prometheus-server-conf
+  namespace: monitoring
+data:
+  prometheus.rules: |-
+    groups:
+    - name: devopscube demo alert
```
Comment (joaquinrz) [inline deploy/ngsadev/monitoring/prometheus.yaml:83]: I don't think we need this "devopscube demo alert" as part of the configmap

File: deploy/ngsadev/monitoring/prometheus.yaml [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+      rules:
+      - alert: High Pod Memory
```
Comment (joaquinrz) [inline deploy/ngsadev/monitoring/prometheus.yaml:85]: same with this alert

File: deploy/ngsadev/monitoring/prometheus.yaml [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+          action: replace
+          target_label: kubernetes_pod_name
+      
+      - job_name: 'kube-state-metrics'
+        static_configs:
+          - targets: ['kube-state-metrics.kube-system.svc.cluster.local:8080']
+      - job_name: 'ngsa-memory'
+        static_configs:
+          - targets: [ 'ngsa-memory.default.svc.cluster.local:8080' ]
+      - job_name: 'ngsa-cosmos'
```
Comment (joaquinrz) [inline deploy/ngsadev/monitoring/prometheus.yaml:158]: might be worth to also add ngsa-java

File: templates/prometheus.yaml [hunk 4]
```diff
@@ -0,0 +1,294 @@
+apiVersion: v1
```
Comment (bartr) [inline templates/prometheus.yaml:1]: I would move the namespace into a separate file - it's in two locations

File: deploy/ngsadev/monitoring/prometheus.yaml (hunk 1 above)
Comment (joheec) [reply deploy/ngsadev/monitoring/prometheus.yaml:83]: done!

File: deploy/ngsadev/monitoring/prometheus.yaml (hunk 2 above)
Comment (joheec) [reply deploy/ngsadev/monitoring/prometheus.yaml:85]: done!

File: deploy/ngsadev/monitoring/prometheus.yaml (hunk 3 above)
Comment (joheec) [reply deploy/ngsadev/monitoring/prometheus.yaml:158]: done!

File: templates/prometheus.yaml (hunk 4 above)
Comment (joheec) [reply templates/prometheus.yaml:1]: I hope I handled this in an acceptable way

**Review by joaquinrz — APPROVED** [review]

**Review by jofultz — APPROVED** [review]

**Pull request merged**